### ISSUE TITLE
custom grains from `base/_grains` folder silently unavailable

### ISSUE THREAD

**Issue by alexykot** [saltstack/salt#35079]
### Description of Issue/Question
I've created two simple custom grains, `lan_ip` and `wan_ip`, it's parsing list of network interfaces and return minion's LAN or WAN IPs respentively. I've put these grains into `/srv/salt/base/_grains` on salt master and run `salt \* saltutil.sync_all`, which among other things returned:

```
minion-name:
    ----------
...
    grains:
        - grains.__init__
        - grains.lan_ip
        - grains.test_grain
        - grains.wan_ip
...
```

Also I have observed custom grains appearing in paths `/var/cache/salt/minion/files/base/_grains` and `/var/cache/salt/minion/extmods/grains` on the minion. 

After that I have also manually restarted minion, just in case, as there is another open bugreport on that. 
But still when I call `salt 'minion-name' grains.get lan_ip` I get nothing back.
Also custom grains are not listed when I call `salt 'minion-name' grains.ls`. No idea why. Also I see nothing relevant in master or minion logs when I try to call those grains.

The `lan_ip.py` grain source:
```
python
#!/usr/bin/env python
import netifaces

def lan_ip():
    interfaces = netifaces.interfaces()
    for interface_name in interfaces:
        if interface_name != 'lo':
            interface_ip = netifaces.ifaddresses(interface_name)[netifaces.AF_INET][0]['addr']
            if (interface_ip.startswith('10.') or interface_ip.startswith('172.16.')
                or interface_ip.startswith('192.168.')):
                return {'lan_ip': interface_ip}

    return {'lan_ip': False}

```

### Setup
Salt-master is installed inside docker container if that makes any difference, and inside container the whole salt states are available as: 
```
file_roots:
  base:
    - /srv/salt/base
  prod:
    - /srv/salt/prod
  stage:
    - /srv/salt/stage
  dev:
    - /srv/salt/dev
```
Custom grains itself are in `/srv/salt/base/_grains`. 

### Versions Report
CentOS 7.1 everywhere.
salt 2016.3.1 (Boron)
salt-minion 2016.3.1 (Boron)

**Comment by Ch3LL** [comment]
@alexykot one good way to debug custom grains is to run command locally on the minion to see if its loading `salt-call -ldebug --local grains.get lan_ip` and you should see any errors in that output. One thing i noticed when attempting to use your grain is this error:

```
[DEBUG   ] Failed to import grains lan_ip:
Traceback (most recent call last):
  File "/home/ch3ll/git/salt/salt/loader.py", line 1296, in _load_module
    ), fn_, fpath, desc)
  File "/var/cache/salt/minion/extmods/grains/lan_ip.py", line 2, in <module>
    import netifaces
ImportError: No module named netifaces
```

You will need to make sure `netifaces` is installed on any minion that you would want to use the custom grain on. Once I installed that I was able to run that command accurately and got the ip address back. Does that help your situation?

**Comment by alexykot** [comment]
Thanks @Ch3LL, that was the cause of the problem - `netifaces` didn't get properly provisioned on few hosts and it caused the issue, and I could see the exception when I ran it locally.

Can we somehow funnel exceptions from custom grains into minion log when called from master to make it easier traceable?

**Comment by alexykot** [comment]
Closing this now as solved, and opened #35299 for logging improvement.

**Comment by Ch3LL** [comment]
Glad you go it figured out :+1: I commented on your other issue. We will track it in an issue already open. Thanks for the great detailed issue report!
